src/datetime.rs

```diff
@@ -1579,6 +1598,13 @@ pub mod serde {
             {
                 Ok(None)
             }
+
+            /// Deserialize a timestamp in seconds since the epoch
+            fn visit_unit<E>(self) -> Result<Option<DateTime<Utc>>, E>
+                where E: de::Error
+            {
+                Ok(None)
+            }
```

Comment:
@quodlibetor, the above test now demonstrates the issue reproducibly.
To trigger the issue it is necessary to match an untagged enum representation.
Removing these lines makes the test fail:
```
$ cargo test --features serde
# [...]
test src/datetime.rs - datetime::serde::ts_milliseconds_option::deserialize (line 1537) ... FAILED
#[...]
failures:

---- src/datetime.rs - datetime::serde::ts_milliseconds_option::deserialize (line 1537) stdout ----
Test executable failed (exit code 101).

stderr:
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("data did not match any variant of untagged enum E", line: 0, column: 0)', src/datetime.rs:33:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace



failures:
    src/datetime.rs - datetime::serde::ts_milliseconds_option::deserialize (line 1537)

test result: FAILED. 243 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```

Reply:
huh, wild, I can now reproduce both the failure and success with your changes. Thanks!

Reply:
Thanks for your support!